CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java

```diff
@@ -0,0 +1,235 @@
+        installedFileLocatorMocked = mock(InstalledFileLocator.class);
+        installedFileLocatorMockedStatic = Mockito.mockStatic(InstalledFileLocator.class);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        installedFileLocatorMockedStatic.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        installedFileLocatorMockedStatic.when(InstalledFileLocator::getDefault)
+                .thenReturn(installedFileLocatorMocked);
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault()
+                .locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false))
+                .thenReturn(iconFile);
+        items.clear();
+        items.add("Item 1");
+        items.add("Item 2");
+        items.add("Item 3");
+        items.add("Item 4");
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+        installedFileLocatorMockedStatic.reset();
+    }
+
+    /**
+     * Test the constructor works with a Text
+     */
+    @Test
+    public void testConstructor_WithText() {
+        final File iconFile = new File("non_existing_icon.png");
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault().locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false)).thenReturn(iconFile);
+
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        String expResult = "Text";
+
+        String result = instance.getText();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test the constructor works with an Icon
+     */
+    @Test
+    public void testConstructor_WithIcon() {
+        final Icon icon = new ImageIcon(COOKIE_ICON_PATH);
+        JDropDownMenu instance = new JDropDownMenu(icon, items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
+    }
+
+    /**
+     * Test of getText method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetText() {
+        System.out.println("getText");
+        String text = "Test text";
+        JDropDownMenu instance = new JDropDownMenu(text, items);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of setText method, of class JDropDownMenu.
+     */
+    @Test
+    public void testSetText() {
+        System.out.println("setText");
+        String text = "Test text";
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        instance.setText(text);
+        String result = instance.getText();
+        assertEquals(result, text);
+    }
+
+    /**
+     * Test of getIcon method, of class JDropDownMenu.
+     */
+    @Test
+    public void testGetIcon() {
+        System.out.println("getIcon");
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        Icon result = instance.getIcon();
```

Comment:
Also checking the Icon is the proper icon here

Reply:
Here they appeared to be equal but none of the assert methods asserted. So I ended up checking the existence.

Reply:
It works in the `testSetIcon` for whatever the Icon we set. So I guess this is fine, we can even get rid of this test...

Reply:
This method is actually covered by `testSetIcon` and `testConstructor_WithIcon` in two different ways. Should I copy paste one of them here or just remove the test?